### PR TITLE
Setup pre-commit hooks

### LINKED ISSUE #22: Set up pre-commit hooks
Going off of [#15](https://github.com/UCI-HyperXite/hx9-control-system/pull/15) to further improve developer experience, we can have the formatting and linting commands run before commits (known as pre-commit hooks) to detect any issues with types or formatting.

This prevents scenarios where a developer would push code to a branch that currently has a pull request open that causes the workflows to raise warnings and/or errors. The developer would either have to force push or make another commit to fix the issue. A pre-commit hook would prevent this entirely by forcing the developer to fix said issues before even pushing to the remote branch.

The following should be run in the pre-commit hooks:
- Prettier
- ESLint
- `tsc` TypeScript compiler
- `cargo run`
- `cargo fmt`
- `cargo clippy`

### PR DESCRIPTION
**Pull request by samderanova**
This PR uses [`pre-commit`](https://www.npmjs.com/package/pre-commit) to set up pre-commit hooks for the following:

- `control-station/`
    - Prettier
    - ESLint
    - `tsc`
 - `pod-operation/`
    - `cargo fmt`
    - `cargo check`
    - `cargo clippy`

### CODE REVIEW

File: package.json
Comment (taesungh) [inline package.json:1]: Chore: Could we reformat this to tabs

File: package.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{
+  "name": "hx9-control-system",
+  "version": "1.0.0",
+  "description": "The control system consists of the pod operation backend and control station client. See the respective README files for setup instructions.",
+  "main": "index.js",
+  "dependencies": {
+    "husky": "^9.0.11"
+  },
```
Comment (taesungh) [inline package.json:8]: Chore: husky is unused?

File: package.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  "devDependencies": {
+    "pre-commit": "^1.2.2"
+  },
+  "scripts": {
+    "tsc": "cd control-station && tsc --noEmit && exit 0",
+    "eslint": "cd control-station && npx eslint . && exit 0",
+    "prettier": "cd control-station && prettier --write . && exit 0",
+    "cargoclippy": "cd pod-operation && cargo clippy && exit 0",
+    "cargocheck": "cd pod-operation && cargo check && exit 0",
+    "cargofmt": "cd pod-operation && cargo fmt && exit 0"
```
Comment (taesungh) [inline package.json:18]: Suggestion: could we reorganize these into logical categories per stack and delegate to the associated app? For example, have a `format` and `lint` script here which delegates to subsequent matching scripts in `control-station/package.json`, similar to a monorepo pipeline.

File: package.json [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  },
+  "pre-commit": ["prettier", "cargofmt"],
```
Comment (taesungh) [inline package.json:20]: Question: do we need to add the other scripts to this array?

**Review by taesungh — CHANGES_REQUESTED** [review]
Nice start, mainly one suggestion about structuring the npm scripts to use corresponding scripts for each app.

File: package.json (hunk 1 above)
Comment (samderanova) [reply package.json:8]: Will remove

File: package.json (hunk 3 above)
Comment (samderanova) [reply package.json:20]: Yes, I will be adding them.

File: package.json [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+		"test:cs": "cd control-station && npm run test",
+		"test:po": "cd pod-operation && npm run test"
+	},
+	"pre-commit": [
+		"format:cs",
+		"format:po",
+		"lint:cs",
+		"lint:po",
+		"test:po"
+	],
```
Comment (taesungh) [inline package.json:22]: Chore: think we also need to include `"test:cs"` here?

File: .gitignore [hunk 5]
```diff
@@ -0,0 +1 @@
+node_modules/
```
Comment (taesungh) [inline .gitignore:1]: Nitpick: could we get a new line at EOF here

File: package.json (hunk 4 above)
Comment (samderanova) [reply package.json:22]: Sure, but do we have any tests for the frontend?

Comment (samderanova) [reply package.json:22]: Oh wait never mind

File: control-station/package.json [hunk 6]
```diff
@@ -7,6 +7,8 @@
 		"dev": "vite",
 		"build": "tsc && vite build",
 		"lint": "eslint . --ext ts,tsx --report-unused-disable-directives --max-warnings 0",
```
Comment (SM-dot) [inline control-station/package.json:9]: Thanks for working on this! Quick question - Is there a specific reason for setting the 'max-warnings' option to 0 in the 'lint' script?

Comment (taesungh) [reply control-station/package.json:9]: This lint script came from the Vite template during project initialization, but to answer, setting the threshold ensures any warnings cause the `eslint` program to error as opposed to announcing the error but still finishing successfully (exit code 0). We want the lint check to fail on warnings accordingly. Refer to [`--max-warnings` in the ESLint CLI Reference](https://eslint.org/docs/latest/use/command-line-interface#--max-warnings) for more detail.

**Review by taesungh — APPROVED** [review]
Thanks for setting all of this up. Side note: pre-commit seems to not be maintained anymore, but we'll stick with this for now. We could maybe also look into using Turborepo in the future for simpler script delegation. Please squash during merge.

**Review by SM-dot — APPROVED** [review]

File: package.json
Comment (SM-dot) [inline package.json:1]: Looks good! Just for clarification "pod operation backend"- what exactly are we planning to have for this?

**Pull request merged**